tests/tests/test_mtls.py

```diff
@@ -100,11 +97,13 @@ def hsm_setup(self, pin, ssl_engine_id, device):
         script = f"""\
 #!/bin/bash
 echo "module: /usr/lib/softhsm/libsofthsm2.so" > /usr/share/p11-kit/modules/softhsm2.module
-mkdir -p /softhsm/tokens
-echo "directories.tokendir = /softhsm/tokens" > /softhsm/softhsm2.conf
-export SOFTHSM2_CONF=/softhsm/softhsm2.conf;
-softhsm2-util --init-token --free --label unittoken1 --pin {pin} --so-pin 0002
-pkcs11-tool --module /usr/lib/softhsm/libsofthsm2.so --login --pin {pin} --write-object "{key}" --type privkey --id 0909 --label privatekey
+softhsm2-util --init-token --free --label testtoken --pin {pin} --so-pin 0002
+pkcs11-tool --module /usr/lib/softhsm/libsofthsm2.so --login --pin {pin} --write-object "{key}" --type privkey --id 0909 --label certificatekey
+
+# Add the public key to the slot also
+# (Needed for the pkcs11-provider - since it does not extract it when needed)
+openssl pkey -in {key} -pubout -out {key}.pub
+pkcs11-tool --module /usr/lib/softhsm/libsofthsm2.so --login --pin {pin} --write-object "{key}.pub" --type pubkey --id 0909 --label certificatekey
```

Comment:
we discussed that over slack. all I can add is: the fact that we need to change the test in such fundamental manner means that the code behaves differently. I find it disturbing. what if your secure element does not let you store the public key?

Reply:
Then it won't work with the `v3 provider` we are using atm

Reply:
This is not a limitation in _our_ code

Reply:
this is highly misleading. please change the description of the commit. it is a different test, testing different code, using different API.